### DISCUSSION TITLE
passing Label as argument to function corresponding to option menu

### DISCUSSION THREAD

**Discussion by DarkKnight263** [TomSchimansky/CustomTkinter#873] [Q&A] [answered]
i am unable to understand what im doing wrong . i expect the label to update dynamically whenever relevant option is selected.

`import tkinter
import tkinter.messagebox
import customtkinter
from PIL import Image

customtkinter.set_appearance_mode("System")  # Modes: "System" (standard), "Dark", "Light"
customtkinter.set_default_color_theme("blue")  # Themes: "blue" (standard), "green", "dark-blue"
cart=[]
class App(customtkinter.CTk):
    
    def __init__(self):
        super().__init__()

        

        # configure window
        self.title("hi guys")
        self.geometry(f"{1100}x{580}")

        # configure grid layout (4x4)
        self.grid_columnconfigure(1, weight=1)
        self.grid_columnconfigure((2, 3), weight=0)
        self.grid_rowconfigure((0, 1, 2), weight=1)

        self.optionmenu_1 = customtkinter.CTkOptionMenu(self, values=["Value 1", "Value 2", "Value 3"],command= lambda: self.abc(self.lb2)).pack()
        self.lb2 = customtkinter.CTkLabel(self, text="hi", anchor="w",font=("HGGothicE",15))
        self.lb2.pack(pady=(10, 10), padx=20)

    def abc(self,choice: str,lb):
        if(choice == "Value 2"):
           lb.configure(text="done")
            
if __name__ == "__main__":
    app = App()
    app.mainloop()`

**Comment by abhaymakes** [comment, marked as answer]
Hey, @DarkKnight263 , here's how you can update the label once an option is selected : 

```python
def __init__(self):
    super().__init__()

    # configure window
    self.title("hi guys")
    self.geometry(f"{1100}x{580}")

    # configure grid layout (4x4)
    self.grid_columnconfigure(1, weight=1)
    self.grid_columnconfigure((2, 3), weight=0)
    self.grid_rowconfigure((0, 1, 2), weight=1)

    self.optionmenu_1 = customtkinter.CTkOptionMenu(self, values=["Value 1", "Value 2", "Value 3"], command=abc).pack()

    self.lb2 = customtkinter.CTkLabel(self, text="hi", anchor="w",font=("HGGothicE",15))
    self.lb2.pack(pady=(10, 10), padx=20)

def abc(self, choice : str):
    if(choice == "Value 2"):
       self.lb2.configure(text="done")
```

Hope it helps!

**Reply by DarkKnight263** [reply]
Thanks a lot man, really appreciate u replying

**Reply by abhaymakes** [reply]
Glad i could help 😁
